webapp/channels/src/components/post_view/reaction_list/reaction_list.tsx

```diff
@@ -78,15 +83,13 @@ export default class ReactionList extends React.PureComponent<Props, State> {
     }
 
     static getDerivedStateFromProps(props: Props, state: State): Partial<State> | null {
-        let emojiNames = state.emojiNames;
+        const emojiNames = Object.values(props.reactions ?? {}).map((reaction) => reaction.emoji_name);
```

Comment:
It turns out that these changes are actually the cause of the failing E2E test. The previous version of the code is deduplicating the names in `emojiNames`, but this doesn't do that. You get some funny side effects with this like there being one reaction shown for every user who reacts with that emoji even though the counters still increment for each one
![Screenshot 2024-09-26 at 1 21 48 PM](https://github.com/user-attachments/assets/ccb259c4-ec99-4cad-9ba3-5f41659ca670)
![Screenshot 2024-09-26 at 1 21 53 PM](https://github.com/user-attachments/assets/67ba52ea-56db-4a2e-8824-8252066dd5e6)

I think we might be fine to just revert the `getDerivedStateFromProps` at this point since I don't think we need to trim `emojiNames` if someone decreases the setting's value